src/VisualStudio/TestUtilities/InProcess/InteractiveWindow_InProc.cs

```diff
@@ -0,0 +1,173 @@
+        /// </summary>
+        public string GetLastReplOutput()
+        {
+            // TODO: This may be flaky if the last submission contains ReplPromptText
+
+            var replText = GetReplTextWithoutPrompt();
+            int lastPromptIndex = replText.LastIndexOf(ReplPromptText);
+
+            replText = replText.Substring(lastPromptIndex, replText.Length - lastPromptIndex);
+            int lastSubmissionIndex = replText.LastIndexOf(ReplSubmissionText);
+
+            if (lastSubmissionIndex > 0)
+            {
+                replText = replText.Substring(lastSubmissionIndex, replText.Length - lastSubmissionIndex);
+            }
+            else if (!replText.StartsWith(ReplPromptText))
+            {
+                return replText;
+            }
+
+            int firstNewLineIndex = replText.IndexOf(Environment.NewLine);
+
+            if (firstNewLineIndex <= 0)
+            {
+                return replText;
+            }
+
+            firstNewLineIndex += Environment.NewLine.Length;
+            return replText.Substring(firstNewLineIndex, replText.Length - firstNewLineIndex);
+        }
+
+        public void Reset(bool waitForPrompt = true)
+        {
+            ExecuteCommand(ResetCommand);
+
+            if (waitForPrompt)
+            {
+                WaitForReplPrompt();
+            }
+        }
+
+        public void SubmitText(string text, bool waitForPrompt = true)
+        {
+            _interactiveWindow.SubmitAsync(new[] { text }).Wait();
+
+            if (waitForPrompt)
+            {
+                WaitForReplPrompt();
+            }
+        }
+
+        public void CloseWindow()
+        {
+            var dte = GetDTE();
+
+            foreach (EnvDTE.Window window in dte.Windows)
+            {
+                if (window.Caption == _windowTitle)
+                {
+                    window?.Close();
+                    break;
+                }
+            }
+        }
+
+        public void ShowWindow(bool waitForPrompt = true)
+        {
+            ExecuteCommand(_viewCommand);
+
+            if (waitForPrompt)
+            {
+                WaitForReplPrompt();
+            }
+        }
+
+        public void WaitForReplPrompt()
+        {
+            WaitForReplPromptAsync().ConfigureAwait(continueOnCapturedContext: false).GetAwaiter().GetResult();
+        }
+
+        private async Task WaitForReplPromptAsync()
+        {
+            while (!GetReplText().EndsWith(ReplPromptText))
+            {
+                await Task.Yield();
+            }
+        }
+
+        public void WaitForReplOutput(string outputText)
+        {
+            WaitForReplOutputAsync(outputText).ConfigureAwait(continueOnCapturedContext: false).GetAwaiter().GetResult();
+        }
+
+        private async Task WaitForReplOutputAsync(string outputText)
+        {
+            while (!GetReplText().EndsWith(outputText + Environment.NewLine + ReplPromptText))
```

Comment:
Yes, we should definitely wait a bit (say, 50 milliseconds) before trying again. Thanks!

As far as employing IAsyncListener, I'd really like to, but it's complicated (and maybe impossible) due to the fact that text can appear in the REPL output asynchronously at any time -- even after the REPL prompt has appeared and the window is "available". This happens in one of our REPL tests where clicking a button in a Window results in text being output to the Console, which the REPL captures after the prompt has been displayed. The problem is that, IAsyncListener can't be plugged into the user's code which is executing in the interactive process -- at least, not without modifying the user's code.